lib/settings/pages/post_appearance_settings_page.dart

```diff
@@ -79,6 +80,15 @@ class _PostAppearanceSettingsPageState extends State<PostAppearanceSettingsPage>
   /// When enabled, posts that have been marked as read will be dimmed
   bool dimReadPosts = true;
 
+  /// When enabled, the full date will be shown
+  bool showFullPostDate = false;
+
+  /// The selected date format. This is only used when `showFullPostDate` is enabled
+  DateFormat? selectedDateFormat;
+
+  /// List of available date formats to select from
+  List<DateFormat> dateFormats = [DateFormat.yMMMMd(Intl.systemLocale).add_jm()];
```

Comment:
Is there any concern that `add_jm()` is not taking a locale here? You can call `DateFormat.jm(<locale>)`, but I couldn't find a great way to combine to formats that both take a locale.

Reply:
I _believe_ it takes the locale from the date format? `DateFormat.yMMMMd(<locale>)` and `DateFormat.jm(<locale>)` are just constructors which create a DateFormat object with the applied locale.

`add_jm()` is then just a member function which changes the internal DateFormat pattern. When we finally parse the date format, I think that's when the locale is applied. Also, it seems like by default, it uses `Intl.systemLocale` for the locale if none is passed in.

https://api.flutter.dev/flutter/intl/DateFormat/DateFormat.d.html

Reply:
Sweet, thanks for confirming that!